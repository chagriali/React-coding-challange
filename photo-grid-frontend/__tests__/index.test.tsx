import { render } from "@testing-library/react";
import Home from "@/pages/index";
import axios from "axios";
import { act } from "react-dom/test-utils";

// Mock jest and set the type
jest.mock("axios");
const mockedAxios = axios as jest.MockedFunction<typeof axios>;

describe("index", () => {
  it("Should render correctly", async () => {
    // Provide the data object to be returned
    mockedAxios.mockResolvedValue({
      data: [
        {
          id: "testiId1",
          urls: { small: "/placeholder.png" },
          width: 500,
          height: 500,
        },
        {
          id: "testiId2",
          urls: { small: "/placeholder.png" },
          width: 500,
          height: 500,
        },
      ],
      status: 200,
      statusText: "Ok",
      headers: {},
      config: {},
    });

    let home: any;
    await act(async () => {
      home = render(<Home />);
    });
    expect(home).toMatchSnapshot();
  });
});
