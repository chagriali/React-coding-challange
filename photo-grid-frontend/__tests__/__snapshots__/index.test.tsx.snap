// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col h-screen justify-between"
      >
        <nav
          class="relative flex flex-wrap items-center justify-between px-2 py-3 bg-zinc-900 mb-3"
        >
          <div
            class="container px-4 mx-auto flex flex-wrap items-center justify-between"
          >
            <div
              class="w-full relative flex justify-between lg:w-auto lg:static lg:block lg:justify-start"
            >
              <a
                class="text-sm font-bold leading-relaxed inline-block mr-4 py-2 whitespace-nowrap uppercase text-white"
                href="#pablo"
              >
                React coding challenge
              </a>
            </div>
          </div>
        </nav>
        <div
          class="infinite-scroll-component__outerdiv"
        >
          <div
            class="infinite-scroll-component "
            style="height: auto; overflow: auto;"
          >
            <div
              class="masonry-grid"
            >
              <div
                class="masonry-grid_column"
                style="width: 33.333333333333336%;"
              >
                <div
                  class="pb-4 break-inside"
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="Picture of the author"
                      class="object-contain rounded-lg shadow hover:brightness-50 cursor-zoom-in"
                      data-nimg="responsive"
                      decoding="async"
                      sizes="100vw"
                      src="/_next/image?url=%2Fplaceholder.png&w=3840&q=75"
                      srcset="/_next/image?url=%2Fplaceholder.png&w=640&q=75 640w, /_next/image?url=%2Fplaceholder.png&w=750&q=75 750w, /_next/image?url=%2Fplaceholder.png&w=828&q=75 828w, /_next/image?url=%2Fplaceholder.png&w=1080&q=75 1080w, /_next/image?url=%2Fplaceholder.png&w=1200&q=75 1200w, /_next/image?url=%2Fplaceholder.png&w=1920&q=75 1920w, /_next/image?url=%2Fplaceholder.png&w=2048&q=75 2048w, /_next/image?url=%2Fplaceholder.png&w=3840&q=75 3840w"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/png;base64,undefined); background-position: 0% 0%;"
                    />
                    <noscript />
                  </span>
                </div>
              </div>
              <div
                class="masonry-grid_column"
                style="width: 33.333333333333336%;"
              >
                <div
                  class="pb-4 break-inside"
                >
                  <span
                    style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                    />
                    <img
                      alt="Picture of the author"
                      class="object-contain rounded-lg shadow hover:brightness-50 cursor-zoom-in"
                      data-nimg="responsive"
                      decoding="async"
                      sizes="100vw"
                      src="/_next/image?url=%2Fplaceholder.png&w=3840&q=75"
                      srcset="/_next/image?url=%2Fplaceholder.png&w=640&q=75 640w, /_next/image?url=%2Fplaceholder.png&w=750&q=75 750w, /_next/image?url=%2Fplaceholder.png&w=828&q=75 828w, /_next/image?url=%2Fplaceholder.png&w=1080&q=75 1080w, /_next/image?url=%2Fplaceholder.png&w=1200&q=75 1200w, /_next/image?url=%2Fplaceholder.png&w=1920&q=75 1920w, /_next/image?url=%2Fplaceholder.png&w=2048&q=75 2048w, /_next/image?url=%2Fplaceholder.png&w=3840&q=75 3840w"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/png;base64,undefined); background-position: 0% 0%;"
                    />
                    <noscript />
                  </span>
                </div>
              </div>
            </div>
            <h3>
              Loading ...
            </h3>
          </div>
        </div>
        <footer
          class="p-4 bg-white rounded-lg shadow md:flex md:items-center md:justify-between md:p-6 dark:bg-gray-800"
        >
          <span
            class="text-sm text-gray-500 sm:text-center dark:text-gray-400"
          >
            Â© 2022 . All Rights Reserved.
          </span>
          <ul
            class="flex flex-wrap items-center mt-3 text-sm text-gray-500 dark:text-gray-400 sm:mt-0"
          >
            <li>
              <a
                class="mr-4 hover:underline md:mr-6 "
                href="#"
              >
                About
              </a>
            </li>
            <li>
              <a
                class="hover:underline"
                href="#"
              >
                Contact
              </a>
            </li>
          </ul>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col h-screen justify-between"
    >
      <nav
        class="relative flex flex-wrap items-center justify-between px-2 py-3 bg-zinc-900 mb-3"
      >
        <div
          class="container px-4 mx-auto flex flex-wrap items-center justify-between"
        >
          <div
            class="w-full relative flex justify-between lg:w-auto lg:static lg:block lg:justify-start"
          >
            <a
              class="text-sm font-bold leading-relaxed inline-block mr-4 py-2 whitespace-nowrap uppercase text-white"
              href="#pablo"
            >
              React coding challenge
            </a>
          </div>
        </div>
      </nav>
      <div
        class="infinite-scroll-component__outerdiv"
      >
        <div
          class="infinite-scroll-component "
          style="height: auto; overflow: auto;"
        >
          <div
            class="masonry-grid"
          >
            <div
              class="masonry-grid_column"
              style="width: 33.333333333333336%;"
            >
              <div
                class="pb-4 break-inside"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                  />
                  <img
                    alt="Picture of the author"
                    class="object-contain rounded-lg shadow hover:brightness-50 cursor-zoom-in"
                    data-nimg="responsive"
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fplaceholder.png&w=3840&q=75"
                    srcset="/_next/image?url=%2Fplaceholder.png&w=640&q=75 640w, /_next/image?url=%2Fplaceholder.png&w=750&q=75 750w, /_next/image?url=%2Fplaceholder.png&w=828&q=75 828w, /_next/image?url=%2Fplaceholder.png&w=1080&q=75 1080w, /_next/image?url=%2Fplaceholder.png&w=1200&q=75 1200w, /_next/image?url=%2Fplaceholder.png&w=1920&q=75 1920w, /_next/image?url=%2Fplaceholder.png&w=2048&q=75 2048w, /_next/image?url=%2Fplaceholder.png&w=3840&q=75 3840w"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/png;base64,undefined); background-position: 0% 0%;"
                  />
                  <noscript />
                </span>
              </div>
            </div>
            <div
              class="masonry-grid_column"
              style="width: 33.333333333333336%;"
            >
              <div
                class="pb-4 break-inside"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                  />
                  <img
                    alt="Picture of the author"
                    class="object-contain rounded-lg shadow hover:brightness-50 cursor-zoom-in"
                    data-nimg="responsive"
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fplaceholder.png&w=3840&q=75"
                    srcset="/_next/image?url=%2Fplaceholder.png&w=640&q=75 640w, /_next/image?url=%2Fplaceholder.png&w=750&q=75 750w, /_next/image?url=%2Fplaceholder.png&w=828&q=75 828w, /_next/image?url=%2Fplaceholder.png&w=1080&q=75 1080w, /_next/image?url=%2Fplaceholder.png&w=1200&q=75 1200w, /_next/image?url=%2Fplaceholder.png&w=1920&q=75 1920w, /_next/image?url=%2Fplaceholder.png&w=2048&q=75 2048w, /_next/image?url=%2Fplaceholder.png&w=3840&q=75 3840w"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/png;base64,undefined); background-position: 0% 0%;"
                  />
                  <noscript />
                </span>
              </div>
            </div>
          </div>
          <h3>
            Loading ...
          </h3>
        </div>
      </div>
      <footer
        class="p-4 bg-white rounded-lg shadow md:flex md:items-center md:justify-between md:p-6 dark:bg-gray-800"
      >
        <span
          class="text-sm text-gray-500 sm:text-center dark:text-gray-400"
        >
          Â© 2022 . All Rights Reserved.
        </span>
        <ul
          class="flex flex-wrap items-center mt-3 text-sm text-gray-500 dark:text-gray-400 sm:mt-0"
        >
          <li>
            <a
              class="mr-4 hover:underline md:mr-6 "
              href="#"
            >
              About
            </a>
          </li>
          <li>
            <a
              class="hover:underline"
              href="#"
            >
              Contact
            </a>
          </li>
        </ul>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
